app/src/main/java/io/intrepid/contest/screens/contestcreation/namecontest/NameContestPresenter.java

```diff
@@ -13,15 +14,34 @@
     }
 
     @Override
+    public void onViewCreated() {
+        super.onViewCreated();
+            view.disableNext();
+    }
+
+    @Override
     public void onContestNameUpdate(String contestName) {
-        if(contestName.isEmpty()){
-           showError();
-        }else{
-          view.saveEnteredName(contestName);
+        view.saveEnteredName(contestName);
+    }
+
+    @Override
+    public void onNextInvalidated() {
+        view.disableNext();
+    }
+
+    @Override
+    public void onNextValidated() {
+        if (view != null) {
+            view.enableNext();
         }
     }
 
-    private void showError() {
-        view.showError();
+    @Override
+    public void onTextChanged(CharSequence newName) {
+        if (TextUtils.isEmpty(newName)) {
```

Comment:
On these two we can also just pass in the newName as a string and then check if its empty in the presenter.